_posts/2018-12-19-bazel-0.21.md

```diff
@@ -0,0 +1,117 @@
+[breaking-change-0.22](https://github.com/bazelbuild/bazel/issues?q=is%3Aissue+label%3Abreaking-change-0.22).
+
+
+## Android
+
+*   [`--experimental_check_desugar_deps`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--experimental_check_desugar_deps)
+    is enabled by default. This flag rules out several types of invalid Android
+    builds at compile-time. Note that this is incompatible with remote Android
+    builds, so you can disable it with `--noexperimental_check_desugar_deps`.
+*   Bazel now supports Android Databinding v2. It can be enabled with
+    [`--experimental_android_databinding_v2`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--experimental_android_databinding_v2).
+
+
+## Apple
+
+*   `XCRUNWRAPPER_LABEL` is removed. If you used this value before, please use @bazel_tools//tools/objc:xcrunwrapper instead.
+
+
+## C++
+
+*   The `--experimental_stl` command line option is removed.
+*   [`cpp` configuration
+    fragment](https://docs.bazel.build/versions/master/skylark/lib/cpp.html)
+    doesn't do package loading anymore. That means:
+    *   You can now use Bazel without a C++ toolchain (as long as you don't build C++ rules).
+    *   Bazel may start about 2 seconds faster than before, because it does not analyze C++ toolchain anymore when it is not needed.
+
+## Java
+
+*   `--strict_android_deps` and `--strict_java_deps` are renamed to `--experimental_strict_java_deps`. (The two flags used to do the same.)
+
+## Starlark
+
+*   A new incompatible flag [`--incompatible_strict_argument_ordering`](https://github.com/bazelbuild/bazel/issues/6611) is added. This concerns a corner case in function calls. When this flag is enabled, `*arg` and `**kwarg` have to be after all other arguments.
+
+## External repositories
+
+*   [`repository_ctx.download`](https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#download) and [`repository_ctx.download_and_extract`](https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#download_and_extract) now return a struct, instead of `None`.
+
+
+## Execution
+
+*   `--genrule_strategy` is deprecated in favor of [`--strategy=Genrule=<strategy>`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--strategy) (for genrules) or [`--spawn_strategy=<strategy>`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--spawn_strategy) (for all actions).
+*   Dynamic execution is now available with [`--experimental_spawn_scheduler`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--experimental_spawn_scheduler). Dynamic execution allows a build action to run locally and remotely simultaneously, and Bazel picks the fastest action. This provides the best of both worlds: faster clean builds than pure local builds, and faster incremental builds than pure remote builds.
+
+
+## Other changes
```

Comment:
Please also mention that the docs are now versioned, thanks to @jin 's much-appreciated efforts \o/